test/solc/CommandLineParser.cpp

```diff
@@ -426,6 +426,58 @@ BOOST_AUTO_TEST_CASE(invalid_options_input_modes_combinations)
 		}
 }
 
+BOOST_AUTO_TEST_CASE(default_optimiser_sequence)
+{
+	auto const& commandLineOptions = parseCommandLine({"solc", "contract.sol", "--optimize"});
```

Comment:
```suggestion
	CommandLineOptions const& commandLineOptions = parseCommandLine({"solc", "contract.sol", "--optimize"});
```

Reply:
Don't really have a problem with this, but curious to know why? This seems like a textbook use of `auto` to me.

Reply:
We have/had a lot of in-fighting about when to use ``auto`` :-). Glad to see that another reasonable position has joined us :-D (I've always advocated using ``auto`` extensively - but we also have people who prefer seeing the types of things locally and I gave into that...)
Or a different way to put it: It was Chris all along :-D.

Reply:
But yeah, the vague policy we have in the code right now is to only use ``auto`` if the type appears in the same statement (like you have a ``dynamic_cast`` or the like on the rhs anyways mentioning the type), if the type is unnamable or on structured binding (i.e. where you have to).

The coding style has ``9. Only use ``auto`` if the type is very long and rather irrelevant.`` which is also pretty vague, but yeah, what I just wrote above is how we ended up doing this, even though we were never unanimous about it - and it's not like I don't see the point of that position at all, so...

Reply:
Yeah, I'll change it.

Reply:
For me the thing is that `auto` helps mostly when writing, not when reading. When reading I almost always prefer to see the type, especially when it's short and using auto would not save much typing anyway.

This is a bit borderline case since the variable is already called `commandLineOptions` but names are not as reliable as just seeing the type.